src/main/java/org/springframework/data/couchbase/core/query/Query.java

```diff
@@ -40,6 +51,20 @@ public Query addCriteria(QueryCriteria criteriaDefinition) {
 		return this;
 	}
 
+	public Query setParameters(JsonValue parameters) {
```

Comment:
I think we should make the parameters the same types we expose on the sdk itself for flexibility (and just store it as an object)